azure/storage/client.go

```diff
@@ -56,28 +56,28 @@ type StorageServiceError struct {
 
 // NewBasicClient constructs a StorageClient with given storage service name
 // and key.
-func NewBasicClient(accountName, accountKey string) (*StorageClient, error) {
+func NewBasicClient(accountName, accountKey string) (StorageClient, error) {
 	return NewClient(accountName, accountKey, DefaultBaseUrl, DefaultApiVersion, defaultUseHttps)
 }
 
 // NewClient constructs a StorageClient. This should be used if the caller
 // wants to specify whether to use HTTPS, a specific REST API version or a
 // custom storage endpoint than Azure Public Cloud.
-func NewClient(accountName, accountKey, blobServiceBaseUrl, apiVersion string, useHttps bool) (*StorageClient, error) {
+func NewClient(accountName, accountKey, blobServiceBaseUrl, apiVersion string, useHttps bool) (StorageClient, error) {
 	if accountName == "" {
-		return nil, fmt.Errorf("azure: account name required")
+		return StorageClient{}, fmt.Errorf("azure: account name required")
```

Comment:
feel free to have `var c StorageClient` in the beginning and you can return that in the errors.